spec/models/person_spec.rb

```diff
@@ -267,7 +267,148 @@ def create(attrs = {})
+        nicknamed_person.last_name = 'Tester'
+
+        expect(nicknamed_person).to be_valid
+      end
+
+      it 'has not only a last_name' do
+        nicknamed_person.first_name = nil
+        nicknamed_person.last_name = nil
+
+        expect(nicknamed_person.first_name).to be_blank
+        expect(nicknamed_person.last_name).to be_blank
+        expect(nicknamed_person).to_not be_valid
+
+        nicknamed_person.first_name = nil
+        nicknamed_person.last_name = 'Tester'
+
+        expect(nicknamed_person).to_not be_valid
+      end
+
+      it 'has not only a first_name' do
+        nicknamed_person.first_name = nil
+        nicknamed_person.last_name = nil
+
+        expect(nicknamed_person.first_name).to be_blank
+        expect(nicknamed_person.last_name).to be_blank
+        expect(nicknamed_person).to_not be_valid
+
+        nicknamed_person.first_name = 'Tom'
+        nicknamed_person.last_name = nil
+
+        expect(nicknamed_person).to_not be_valid
+      end
+
+      it 'has an address' do
+        previous_value = named_person.address
+        named_person.address = nil
+
+        expect(named_person.address).to be_blank
+        expect(named_person).to_not be_valid
+
+        named_person.address = previous_value
+
+        expect(named_person).to be_valid
+      end
+
+      it 'has a zip_code' do
+        previous_value = named_person.zip_code
+        named_person.zip_code = nil
+
+        expect(named_person.zip_code).to be_blank
+        expect(named_person).to_not be_valid
+
+        named_person.zip_code = previous_value
+
+        expect(named_person).to be_valid
+      end
+
+      it 'has a town' do
+        previous_value = named_person.town
+        named_person.town = nil
+
+        expect(named_person.town).to be_blank
+        expect(named_person).to_not be_valid
+
+        named_person.town = previous_value
+
+        expect(named_person).to be_valid
+      end
+
+      it 'has a country' do
+        previous_value = named_person.country
+        named_person.country = nil
+
+        expect(named_person.country).to be_blank
+        expect(named_person).to_not be_valid
+
+        named_person.country = previous_value
+
+        expect(named_person).to be_valid
+      end
+
+      it 'has a correspondence_language' do
+        previous_value = named_person.correspondence_language
+        named_person.correspondence_language = nil
+
+        expect(named_person.correspondence_language).to be_blank
+        expect(named_person).to_not be_valid
+
+        named_person.correspondence_language = previous_value
+
+        expect(named_person).to be_valid
+      end
+
+      it 'has a language' do
+        previous_value = named_person.language
+        named_person.language = nil
+
+        expect(named_person.language).to be_blank
+        expect(named_person).to_not be_valid
+
+        named_person.language = previous_value
+
+        expect(named_person).to be_valid
```

Comment:
Wie wäre es wenn du diese Tests so umschreibst?
```
expect { named_person.some_required_field = nil }.to change { named_person.valid? }.from(true).to(false)
```
Oder falls es dir wichtig ist, die Transition von `nil` auf befüllt zu testen, vielleicht in einem `before` auf `nil` setzen und das assignment abändern?

Reply:
Danke für diesen Hinweis, das liest sich doch gleich viel deutlicher.

Die Richtung, dass ein valider Record invalide wird, reicht mir völlig. Immerhin zeigt das die Validation doch recht gut. Ich habe miene Specs mal entsprechende verkürzt. Da fragt man sich doch direkt, ob dieser change matcher nicht irgendwo als `invalidate_record(named_person)` zu haben ist :-)